### PR TITLE
[aosp/LA.UM.7.1.r1] Add option to set build_directory manually

### PR DESCRIPTION
**Pull request by Thaodan**

### CODE REVIEW

**Comment by MarijnS95** [comment]
It's pretty common (make, cmake, ninja, tar, unzip etc) to use `-C` to specify an output directory (usually interpreted as "change to this directory before running).

**Comment by Thaodan** [comment]
No -C is to switch to that directory: "Change to directory dir before reading the makefiles or doing anything else"

**Comment by sledges** [comment]
And when you switch to an empty directory, the build should then fail, so using `-C` doesn't really fit here.

May I propose a compromise, `make O=DIR` is what we're trying to achieve here, so maybe an `-O`? :)

PS: tested the PR, works as expected!

**Comment by Thaodan** [comment]
Sounds good to me,  adjusted to that idea.

File: build_shared_vars.sh [hunk 1]
```diff
@@ -24,11 +24,12 @@ EOF
 }
```
Comment (sledges) [inline build_shared_vars.sh:25]: ^ usage would need a new parameter too

**Review by sledges — CHANGES_REQUESTED** [review]
May I suggest to squash both commit as they are mutually inclusive?

File: build_shared_vars.sh [hunk 2]
```diff
@@ -24,11 +24,12 @@ EOF
...
 
 
-arguments=khd:
+arguments=khd:D:
```
Comment (sledges) [inline build_shared_vars.sh]: This also needs changing to O.

**Review by sledges — CHANGES_REQUESTED** [review]

**Comment by Thaodan** [comment]
Done, done and done

**Pull request merged**